android_client/ChitChat/app/src/main/java/com/moc/chitchat/activity/LoginActivity.java

```diff
@@ -51,20 +66,64 @@ public void onClick(View v) {
     }
 
     private void loginButton() {
+        this.loginController.loginUser(
+            this,
+            this,
+            this,
+            this.usernameField.getText().toString(),
+            this.passwordField.getText().toString()
+        );
     }
 
     private void registerButton() {
         Intent registerIntent = new Intent(this, RegistrationActivity.class);
         startActivity(registerIntent);
+        overridePendingTransition(R.transition.anim_left1,R.transition.anim_left2);
     }
 
+    /**
+     * onErrorResponse handles errors (validation) from the server and displays them to the User.
+     * @param error An error encapsulating the server response.
+     */
     @Override
     public void onErrorResponse(VolleyError error) {
+        System.out.println("Error registering");
+
+        try {
 
+            Toast.makeText(this,
+                String.format("Invalid credentials or you didn't registered yet"),
+                Toast.LENGTH_LONG).show();
+        } catch (Exception e) {
```

Comment:
@aakyol The try/catch here is redundant and so is the String.format :) So this bit could just be:
```
Toast.makeText(this,
            "Invalid credentials or you didn't registered yet",
            Toast.LENGTH_LONG
        ).show();
```